core/src/datayoga_core/expression.py

```diff
@@ -143,10 +143,10 @@ def compile(self, expression: str):
         self.expression = jmespath.compile(expression)
         self.filter_expression = jmespath.compile(f"[?{expression}]")
 
-    def filter(self, data: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    def filter(self, data: Union[Dict[str, Any], List[Dict[str, Any]]]) -> Any:
```

Comment:
Shouldn't it return:
Optional[List[Dict[str, Any]]
?

Reply:
It could return anything depends on call. In tests it returns direct value, for example.